### ISSUE TITLE
Allow to pass IAM role to "put_subscription_filter" call

### ISSUE THREAD

**Issue by dschro-1993** [cloud-custodian/cloud-custodian#7609]
### Describe the feature

Hey guys,

atm we try to setup a logging infra in our AWS ORG with one Firehose stream in a central AWS account.

This means we need to create one CW logs destination per REGION in this central account to be able to forward the logs from all our member accounts / REGIONS to these destinations, respectively to our Firehose stream.

On AWS side this setup is documented here: [Docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/CrossAccountSubscriptions-Firehose.html)

(To add a subscription filter to every lambda-based log group we would like to use your `c7n-logexporter` tool...)

Now let's talk about security:
We would like to add a AccessPolicy to all our destinations which whitelists all our accounts.
Documented here: [Docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/CreateFirehoseStreamDestination.html)

Our issue: Too many accounts, we would definitely exceed the character size limit of our policy...
So we need to whitelist the ORG-ID instead of the accounts...

To be able to do this we need an additional IAM role in all our member accounts that needs to be passed to the CW logs service for the `put_subscription_filter` call.

Documented here: [Docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/CreateSubscriptionFilter-IAMrole.html)
Also have a look here: [Docs](https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutSubscriptionFilter.html)

Your `c7n-logexporter` tool does not support that feature at the moment:
https://github.com/cloud-custodian/cloud-custodian/blob/master/tools/c7n_logexporter/c7n_logexporter/exporter.py#L146-L151

So my question:
Do you want to add this feature on your own? Or do you want me to provide it via PR?

Thanks!

**Comment by kapilt** [comment]
hmm.. I don't recall needing this in log exporter for cross account kinesis delivery (its original use case), but sure happy to have support in their for role arn specification (or just a role name with role arn constructed to avoid the duplication). the docs are pretty unclear regarding specifying the role arn per the api doc link, :/ https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutSubscriptionFilter.html

```
You don't need to provide the ARN when you are working with a logical destination for cross-account delivery.
```

I guess if your going to use an org condition in the destination policy you'll need it else the principal is the service.

wrt to role provisioning, a stack set or incorporating into your account factory is probably best.

wrt to the orgid for destination policy, you probably want to validate the condition key works in the destination policy there re orgperincipal id. if it works, an option to switch from accounts to user specified would also be welcome. I think I last used this with ~200 accounts (its a single statement, so minimized on duplication) with explicit accounts in the destination policy.

**Comment by dschro-1993** [comment]
Hey @kapilt, thx for answer/input. Really appreciate that!

Let me try to be more precise with what we like to achieve/implement.

AWS announced a new feature on Jan 5, 2022 to support ORG-IDs on destination access policies for cross-account delivery.
[Docs](https://aws.amazon.com/about-aws/whats-new/2022/01/amazon-cloudwatch-logs-aws-organizations-subscriptions/)

And this features is what we like to use.

**Why?**
We have thousands of accounts to support/monitor in our AWS-ORG. So when I specify a list of all those accounts as whitelisted principals in the access policy the call fails.

```
aws logs put-destination-policy --destination-name test --access-policy file://test.json
```

Result:

```
An error occurred (InvalidParameterException) when calling the PutDestinationPolicy operation: 1 validation error detected: Value '{
    "Version" : "2012-10-17",
    "Statement" : [
      {
        "Effect" : "Allow",
        "Principal" : {
          "AWS" : ["111111111111", and thousands of other accounts ...' at 'accessPolicy' failed to satisfy constraint: Member must have size less than or equal to 5120
```

So the size is limited to 5120 characters. This is why we need to use the ORG-ID in the condition block of this access policy instead of the list of accounts. This is why I would like to prepare a PR to support this feature if you like to?

Thanks!

**Comment by kapilt** [comment]
merged in #7617
